frontend/src/map.jsx

```diff
@@ -0,0 +1,148 @@
+import { Map, View } from 'ol';
+import { Fill, Stroke, Style } from 'ol/style';
+import MVT from 'ol/format/MVT';
+import OSM from 'ol/source/OSM';
+import Overlay from 'ol/Overlay';
+import React, { useState, useEffect, useRef } from 'react';
+import TileLayer from 'ol/layer/Tile';
+import VectorTileLayer from 'ol/layer/VectorTile';
+import VectorTileSource from 'ol/source/VectorTile';
+
+import Popup from './popup';
+import { getParcelInfo } from './requests';
+
+// style for selected features
+const selectedStyle = new Style({
+  stroke: new Stroke({
+    color: 'rgba(51, 153, 204,0.8)',
+    width: 4,
+  }),
+  fill: new Fill({
+    color: 'rgba(51, 153, 204,0.4)',
+  }),
+});
+
+function getCoords(geometry) {
+  const flatCoords = geometry.getFlatCoordinates();
+  const pairedCoords = flatCoords.reduce(
+    function(result, value, index, array) {
+      if (index % 2 === 0)
+        result.push(array.slice(index, index + 2));
+      return result;
+    }, []);
+  return pairedCoords;
+}
+
+export default function MapComponent(props) {
+
+  const { toggleEditMenu } = props;
+
+  const [popup, setPopup] = useState(undefined);
+
+  // element refs: used to insert openlayers-controlled nodes into the dom
+  const mapElementRef = useRef();
+  const overlayElementRef = useRef();
+
+  // object refs: used to pass values to child components
+  const overlayRef = useRef()
+
+  // initialize map on first render - logic formerly put into componentDidMount
+  useEffect(() => {
+
+    // Create an overlay to anchor the popup to the map.
+    const overlay = new Overlay({
+      element: overlayElementRef.current,
+      autoPan: {
+        animation: {
+          duration: 250,
+        },
+      },
+    });
+    overlayRef.current = overlay
+
+    const parcelLayer = new VectorTileLayer({
+      source: new VectorTileSource({
+        format: new MVT(),
+        // access API key loaded from your private .env file using dotenv package
+        // because of vite, env variables are exposed through import.meta.env
+        // and must be prefixed with VITE_. https://vitejs.dev/guide/env-and-mode.html#env-files
```

Comment:
Nice, this will undoubtedly be useful in the long-term. Thanks for figuring it out!